commit 938b51f416f68283c7eb6932f6903e68c1810758 (refs/changes/48/4748/1)
Author: sjorge <sjorge@blackdot.be>
Date:   2018-08-25T13:35:08+02:00 (1 year, 1 month ago)
    
    OS-XXXX update vmadm manpage to bring it more inline with reality

diff --git a/src/vm/man/vmadm.1m.md b/src/vm/man/vmadm.1m.md
index cd419449..1b601835 100644
--- a/src/vm/man/vmadm.1m.md
+++ b/src/vm/man/vmadm.1m.md
@@ -7,14 +7,15 @@ vmadm(1M) -- Manage SmartOS virtual machines
 ## DESCRIPTION
 
 The vmadm tool allows you to interact with virtual machines on a SmartOS
-system. All 3 of: OS Virtual Machines (SmartOS zones), LX Virtual Machines
-and KVM Virtual Machines can be managed. vmadm allows you to create, inspect,
-modify and delete virtual machines on the local system.
+system. All 4 of: OS Virtual Machines (SmartOS zones), LX Virtual Machines,
+KVM Virtual Machines, and BHYVE Virtual Machine can be managed. vmadm allows 
+you to create, inspect, modify and delete virtual machines on the local system.
 
 IMPORTANT: Support for LX VMs is currently limited and experimental. This means
 it is very likely to change in major ways without notice. Also: not all the LX
 functionality that *is* implemented is documented yet. The documentation will
-be updated as things stabilize.
+be updated as things stabilize. Most properties that apply to OS VMs also
+apply to LX VMs.
 
 The primary reference for a VM is its UUID. Most commands operate on VMs by
 UUID. In SmartOS, there are included bash tab-completion rules so that you can
@@ -518,7 +519,8 @@ tab-complete UUIDs rather than having to type them out for every command.
 
         type -- type of the properties value.
 
-        vmtype -- types of VM (KVM, LX, OS) for which this property applies.
+        vmtype -- types of VM (BHYVE, KVM, LX, OS) for which this property
+                  applies.
 
         listable -- if they can be included in the -o or -s lists for the
                     'vmadm list' command.
@@ -541,7 +543,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         required to be unique.
 
         type: string
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: yes
@@ -574,7 +576,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         reboot into the intended state.
 
         type: boolean
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: yes
@@ -585,11 +587,23 @@ tab-complete UUIDs rather than having to type them out for every command.
         should fall into.
 
         type: string (UUID)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: yes
-        default: 00000000-0000-0000-0000-000000000000
+        defaul:_opts 00000000-0000-0000-0000-000000000000
+
+    bhyve_extra_opts:
+
+        This allows you to specify additional bhyve cmdline arguments, this
+        string (if set) will be appended to the end of the bhyve cmdline. It
+        is intended for debugging and not for general use.
+
+        type: string (space-separated options for bhyve)
+        vmtype: BHYVE
+        listable: no
+        create: yes
+        update: yes
 
     boot:
 
@@ -610,19 +624,34 @@ tab-complete UUIDs rather than having to type them out for every command.
         available, it will indicate the time the VM last booted.
 
         type: string (ISO 8601 timestamp)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: no
         update: no
 
+    bootrom:
+
+        This indicates the bootrom to use for bhyve, valid values are 'bios',
+        'uefi', or a path to a bootrom binary. The path if specified is
+        evaluated within the zoneroot of the VM so /uefi-debug.bin will
+        actually be something like the path /zones/<uuid>/root/uefi-debug.img
+        from the global zone.
+
+        type: string
+        vmtype: BHYVE
+        listable: no
+        create: yes
+        update: yes
+        default: 'bios'
+
     brand:
 
         This will be one of 'joyent', 'joyent-minimal' or 'lx' for OS
         virtualization and 'kvm' for full hardware virtualization. This is a
         required value for VM creation.
 
-        type: string (joyent|joyent-minimal|lx|kvm)
-        vmtype: OS,KVM
+        type: string (joyent|joyent-minimal|lx|kvm|bhyve)
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: no
@@ -634,7 +663,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         Eg. a value of 300 means up to 3 full CPUs.
 
         type: integer (percentage of single CPUs, set to 0 for no cap)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: yes (live update)
@@ -649,7 +678,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         contention.
 
         type: integer (number of shares)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: yes (live update)
@@ -673,7 +702,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         The time at which the VM was created in ISO 8601 format.
 
         type: string (format: '2011-12-31T06:38:42.457Z')
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: no (automatically added)
         update: no
@@ -686,7 +715,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         in the VM, or from another node).
 
         type: string (compute node's UUID)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: no
         create: no
         update: no
@@ -706,7 +735,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         the zone.
 
         type: JSON Object (key: value)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: no
         create: yes
         update: yes (but see special notes on update command)
@@ -770,7 +799,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         volblocksize property set.
 
         type: integer (block size in bytes, 512 to 131072, must be power of 2)
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: no
         create: yes
         update: no (except when adding new disks)
@@ -781,7 +810,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         Specifies whether this disk should be bootable (only one disk should).
 
         type: boolean
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes (special, see description in 'update' section above)
@@ -796,7 +825,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         See zfs_root_compression section below for more details.
 
         type: string one of: "on,off,gzip,gzip-N,lz4,lzjb,zle"
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: no
         create: yes
         update: yes (see caveat in zfs_root_compression section below)
@@ -810,7 +839,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         creating a disk.
 
         type: boolean
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: no
         create: yes
         update: no (except when adding new disks)
@@ -822,7 +851,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         either this and 'image_size' and 'image_uuid', or 'size' for a disk.
 
         type: string
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes (special, see description in 'update' section above)
@@ -839,7 +868,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         include image_uuid for a disk and not allowed when you don't.
 
         type: integer (size in MiB)
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes (special, see description in 'update' section above)
@@ -851,7 +880,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         UUID must show up in the 'imgadm list' output in order to be valid.
 
         type: string (UUID)
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes (special, see description in 'update' section above)
@@ -864,7 +893,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         man page's description of refreservation.
 
         type: integer number of MiB
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: no
         create: yes
         update: yes (special, see description in 'update' section above)
@@ -881,7 +910,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         and not allowed when you do.
 
         type: integer (size in MiB)
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes (special, see description in 'update' section above)
@@ -892,7 +921,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         Specify whether this disk is a 'disk' or 'cdrom'.
 
         type: string (one of ['disk','cdrom'])
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes (special, see description in 'update' section above)
@@ -906,7 +935,7 @@ tab-complete UUIDs rather than having to type them out for every command.
 
         type: string (KVM: one of ['virtio','ide','scsi'])
                      (BHYVE: one of ['virtio','ahci'])
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes (special, see description in 'update' section above)
@@ -917,7 +946,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         The zpool in which to create this zvol.
 
         type: string (zpool name)
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes (special, see description in 'update' section above)
@@ -947,7 +976,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         true. When set false the property will not appear.
 
         type: boolean
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: no
         create: yes
         update: yes
@@ -1079,7 +1108,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         Normally you do *not* want to set this for KVM.
 
         type: string (UUID)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: no
@@ -1099,7 +1128,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         instead of customer_metadata.
 
         type: JSON Object (key: value)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: no
         create: yes
         update: yes (but see special notes on update command)
@@ -1113,7 +1142,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         customer_metadata. They will also be read-only from within the zone.
 
         type: list of strings
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: no
         create: yes
         update: yes
@@ -1162,7 +1191,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         deleted.
 
         type: boolean
-        vmtype: KVM,LX,OS
+        vmtype: KVM,LX,OS,BHYVE
         listable: yes
         create: yes
         update: yes
@@ -1184,7 +1213,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         contains this VM. See privileges(5) for details on possible privileges.
 
         type: string (comma separated list of zone privileges)
-        vmtype: OS,KVM
+        vmtype: OS
         listable: no
         create: yes
         update: yes
@@ -1209,7 +1238,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         this VM. This value cannot be higher than max_physical_memory.
 
         type: integer (number of MiB)
-        vmtype: OS,KVM
+        vmtype: OS
         listable: yes
         create: yes
         update: yes (live update)
@@ -1221,7 +1250,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         running on the host.
 
         type: integer (number of LWPs)
-        vmtype: OS,KVM
+        vmtype: OS
         listable: yes
         create: yes
         update: yes (live update)
@@ -1234,7 +1263,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         ram + 1024.
 
         type: integer (number of MiB)
-        vmtype: OS,KVM
+        vmtype: OS
         listable: yes
         create: yes
         update: yes (live update)
@@ -1246,7 +1275,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         cannot be lower than max_physical_memory, nor can it be lower than 256.
 
         type: integer (number of MiB)
-        vmtype: OS,KVM
+        vmtype: OS
         listable: yes
         create: yes
         update: yes (live update)
@@ -1291,7 +1320,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         required of a DHCP server will not get through.
 
         type: boolean
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1304,7 +1333,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         specified by the ip property.
 
         type: boolean
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1316,7 +1345,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         from this nic with MAC addresses that don't match the mac property.
 
         type: boolean
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1329,7 +1358,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         from this nic.
 
         type: boolean
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1344,7 +1373,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         this option unless you fully understand the security implications.
 
         type: boolean
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1355,7 +1384,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         Array of ports on which this nic is prevented from sending traffic.
 
         type: array
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1370,7 +1399,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         'fe82::/15', '2600:3c00::f03c:91ff:fe96:a267'.
 
         type: array (of IP addresses or CIDR ranges)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1385,7 +1414,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         Identifiers can be listed.
 
         type: array (of even-lengthed hexadecimal strings beginning with "0x")
-        vmtype: OS,LX,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1397,7 +1426,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         required of a DHCP server will not get through.
 
         type: boolean
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1410,7 +1439,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         nics.*.gateways.
 
         type: string (IPv4 address)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1426,7 +1455,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         future to make it easier to add or remove addresses.
 
         type: array (of IPv4 addresses)
-        vmtype: OS,LX,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1437,7 +1466,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         will always be in the format netX where X is an integer >= 0.
 
         type: string (netX)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: no
@@ -1449,7 +1478,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         nics.*.ips.
 
         type: string (IPv4 address or 'dhcp')
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1471,7 +1500,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         future to make it easier to add or remove addresses.
 
         type: array (of IP addresses with routing prefixes, 'dhcp' or 'addrconf')
-        vmtype: OS,LX,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1481,7 +1510,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         MAC address of virtual NIC.
 
         type: string (MAC address)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: no (see 'update' command description)
@@ -1520,7 +1549,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         The netmask for this NIC's network (not required if using DHCP)
 
         type: string (IPv4 netmask, eg. 255.255.255.0)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1530,7 +1559,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         UUID for allowing nics to be tracked in an external system
 
         type: string (UUID)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes
@@ -1542,7 +1571,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         Names' field in `sysinfo`, or an etherstub or device name.
 
         type: string (device name or nic tag name)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update yes (requires zone stop/boot)
@@ -1555,7 +1584,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         to set two nics to primary is an error.
 
         type: boolean (only true is valid)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes (see above)
         create: yes
         update: yes (setting primary=true on one NIC removes the flag from the
@@ -1626,7 +1655,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         be used to tie this system to others.
 
         type: string (UUID)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: yes
@@ -1638,7 +1667,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         Other users can ignore this field.
 
         type: string
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: yes
@@ -1649,7 +1678,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         Other users can ignore this field.
 
         type: string
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: yes
@@ -1660,7 +1689,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         `init` process for the zone.
 
         type: integer (PID)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: no
         update: no
@@ -1702,27 +1731,28 @@ tab-complete UUIDs rather than having to type them out for every command.
         This sets a quota on the zone filesystem. For OS VMs, this value is the
         space actually visible/usable in the guest. For KVM VMs, this value is
         the quota for the Zone containing the VM, which is not directly
-        available to users.
+        available to users. For BHYVE VMs this needs to be big enough for all
+        configured disks as they are children of the zone/<uuid> dataset.
 
         Set quota to 0 to disable (ie. for no quota).
 
         type: integer (number of GiB)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: yes (live update)
 
     ram:
 
-        For KVM VMs this is the amount of virtual RAM that will be available to
-        the guest kernel. For OS VMs this will be the same as the property
-        max_physical_memory.
+        For KVM VMs and BHYVE VMs this is the amount of virtual RAM that will
+        be available to the guest kernel. For OS VMs this will be the same as
+        the property max_physical_memory.
 
         type: integer (number of MiB)
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: yes
-        create: KVM VMs only
-        update: KVM VMs only, for OS VMs update max_physical_memory instead.
+        create: yes
+        update: yes (requires VM reboot to take effect)
         default: 256
 
     resolvers:
@@ -1833,7 +1863,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         See the 'VM STATES' section below for more details.
 
         type: string
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: no
         update: no
@@ -1878,7 +1908,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         of 'running'.
 
         type: string value, one of: ['stopped', 'start', 'running']
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: no
         create: no
         update: no
@@ -1886,11 +1916,11 @@ tab-complete UUIDs rather than having to type them out for every command.
     type:
 
         This is a virtual field and cannot be updated. It will be 'OS' when the
-        brand == 'joyent*', 'LX' when the brand == 'lx', and 'KVM' when the
-        brand == 'kvm'.
+        brand == 'joyent*', 'LX' when the brand == 'lx', 'KVM' when the
+        brand == 'kvm', and 'BHYVE' when the brand == 'bhyve'.
 
-        type: string value, one of: ['OS', 'LX', 'KVM']
-        vmtype: OS,KVM
+        type: string value, one of: ['OS', 'LX', 'KVM', 'BHYVE']
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: no, set by 'brand' property.
         update: no
@@ -1902,7 +1932,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         after a VM is created.
 
         type: string (UUID)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: no
@@ -1913,11 +1943,11 @@ tab-complete UUIDs rather than having to type them out for every command.
         For KVM VMs this parameter defines the number of virtual CPUs the guest
         will see. Generally recommended to be a multiple of 2.
 
-        type: integer (number of CPUs)
-        vmtype: KVM
+        type: integer (number of vCPUs)
+        vmtype: KVM,BHYVE
         listable: yes
-        create: KVM only
-        update: KVM only (requires VM reboot to take effect)
+        create: yes
+        update: yes (requires VM reboot to take effect)
         default: 1
 
     vga:
@@ -1980,7 +2010,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         to disable TCP listening.
 
         type: integer (0 for random, -1 for disabled)
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: no
         create: yes
         update: yes
@@ -2026,7 +2056,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         within the zone. The root user in the GZ is immune to this limit.
 
         type: integer (0+, set to '' or undefined to unset)
-        vmtype: OS,LX
+        vmtype: OS
         listable: no
         create: yes
         update: yes
@@ -2042,7 +2072,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         available IO.
 
         type: integer (relative value)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: yes (live update)
@@ -2101,7 +2131,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         the zone. The root user in the GZ is immune to this limit.
 
         type: integer (0+, set to '' or undefined to unset)
-        vmtype: OS,LX
+        vmtype: OS
         listable: no
         create: yes
         update: yes
@@ -2118,7 +2148,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         relied on for other things at this point.
 
         type: integer (size in bytes)
-        vmtype: OS,LX
+        vmtype: OS
         listable: no
         create: yes
         update: yes
@@ -2150,7 +2180,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         NOTE: currently this should only ever appear for "docker" VMs.
 
         type: string (3 character mode string)
-        vmtype: OS,LX
+        vmtype: OS
         listable: no
         create: no (handled via docker:* metadata)
         update: no (handled via docker:* metadata)
@@ -2164,7 +2194,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         See the 'VM STATES' section below for more details.
 
         type: string
-        vmtype: KVM
+        vmtype: KVM,BHYVE
         listable: yes
         create: no
         update: no
@@ -2177,7 +2207,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         you'll find things such as the logs and sockets for a VM.
 
         type: string (path)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: no
         create: no (automatic)
         update: no
@@ -2189,7 +2219,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         this property with the create payload, but such use is discouraged.
 
         type: string
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes (OS VMs only)
         update: no
@@ -2203,7 +2233,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         VM is started/stopped.
 
         type: integer
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: no
         update: no
@@ -2216,7 +2246,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         whenever the VM is stopped or started. Do not rely on this value.
 
         type: integer
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: no
         update: no
@@ -2228,7 +2258,7 @@ tab-complete UUIDs rather than having to type them out for every command.
         For KVM VMs, this is only used by the zone shell that the VM runs in.
 
         type: string (zpool name)
-        vmtype: OS,KVM
+        vmtype: OS,KVM,BHYVE
         listable: yes
         create: yes
         update: no
